src/nimble/NimbleBluetooth.cpp

```
@@ -84,7 +98,63 @@ static void purgeIncompatibleBleBonds()
+}
+#endif
+
+namespace
+{
+constexpr uint8_t kNimbleInitRetryLimit = 3;
+constexpr uint32_t kNimbleInitRetryBaseDelayMs = 1000;
+bool pendingNimbleInitRetry = false;
+uint8_t nimbleInitRetryCount = 0;
+uint32_t nimbleInitRetryStartedAtMs = 0;
+
+constexpr uint32_t nimbleInitRetryDelayMs(uint8_t attempt)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)